gateway/src/config.rs

```diff
@@ -0,0 +1,535 @@
+                ratelimit_messages: true,
+                session: None,
+                tls: TlsContainer::new().expect("failed to build tls"),
+                token: token.into_boxed_str(),
+            },
+        }
+    }
+
+    /// Create a new builder from an existing configuration.
+    pub const fn with_config(config: Config) -> Self {
+        Self { inner: config }
+    }
+
+    /// Consume the builder, constructing a shard.
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn build(self) -> Config {
+        self.inner
+    }
+
+    /// Set the event types to process.
+    ///
+    /// This is an optimization technique; all events not included in the
+    /// provided event type flags will not be deserialized by the gateway and
+    /// will be discarded.
+    pub const fn event_types(mut self, event_types: EventTypeFlags) -> Self {
+        self.inner.event_types = event_types;
+
+        self
+    }
+
+    /// Set the proxy URL for connecting to the gateway.
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn gateway_url(mut self, gateway_url: String) -> Self {
+        self.inner.gateway_url = Some(gateway_url);
+
+        self
+    }
+
+    /// Set the properties to identify with.
+    ///
+    /// This may be used if you want to set a different operating system, for
+    /// example.
+    ///
+    /// # Examples
+    ///
+    /// Set the identify properties for a shard:
+    ///
+    /// ```no_run
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// use std::env::{self, consts::OS};
+    /// use twilight_gateway::{Config, Intents, Shard};
+    /// use twilight_model::gateway::payload::outgoing::identify::IdentifyProperties;
+    ///
+    /// let token = env::var("DISCORD_TOKEN")?;
+    /// let properties = IdentifyProperties::new("twilight.rs", "twilight.rs", OS);
+    ///
+    /// let config = Config::builder(token, Intents::empty())
+    ///     .identify_properties(properties)
+    ///     .build();
+    /// # Ok(()) }
+    /// ```
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn identify_properties(mut self, identify_properties: IdentifyProperties) -> Self {
+        self.inner.identify_properties = Some(identify_properties);
+
+        self
+    }
+
+    /// Set the maximum number of members in a guild to load the member list.
+    ///
+    /// Default value is `50`. The minimum value is `50` and the maximum is
+    /// `250`.
+    ///
+    /// # Examples
+    ///
+    /// If you pass `200`, then if there are 250 members in a guild the member
+    /// list won't be sent. If there are 150 members, then the list *will* be
+    /// sent.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the provided value is below 50 or above 250.
+    #[track_caller]
+    pub const fn large_threshold(mut self, large_threshold: u64) -> Self {
+        /// Maximum value of an acceptable [large threshold].
+        ///
+        /// [large threshold]: ConfigBuilder::large_threshold
```

Comment:
Intra-link feels excessive. 
```suggestion
        /// Maximally acceptable `large_threshold` value.
```